### ISSUE TITLE
Add multi-GPU and distributed training support

### ISSUE THREAD

**Issue by teleportup** [teleportup/FedLEO-Federated-Learning#7]
Enable training on multiple GPUs and distributed clusters:
- Data parallelism across GPUs
- Gradient accumulation
- Distributed dataloader
- Communication optimization
- Training on cloud infrastructure (AWS/GCP)

Priority: High
Estimate: 12 hours
